src/inventory-staff-order/services/staff-order.service.ts

```diff
@@ -1,18 +1,49 @@
 import httpStatus from "http-status";
 
-import { StaffOrderDTO } from "../DTOs/StaffOrderDTO";
+import { AddStaffOrderDTO } from "../DTOs/StaffOrderDTO";
 import { getUser } from "../../inventory-accounts/user/services";
+import { StaffOrderModel } from "../../inventory-entities/StaffOrder";
+import { ProductModel } from "../../inventory-entities/Product";
+import { NewStockModel } from "../../inventory-entities/NewStock";
+// import { AddNewStockDTO } from "src/inventory-new-stock/DTOs/NewStockDTO";
+// import { multiwait } from "../../inventory-shared/multiwait";
 
 
 
-export const StaffOrderService = async ({staffID, department, branch, inStock, notInStock}: StaffOrderDTO) => {
+export const AddStaffOrderService = async ({staffID, department, branch, inStock, notInStock}: AddStaffOrderDTO) => {
+   
     const user = await getUser({userID: staffID});
-    if (!user) return {success: false, status: httpStatus.NOT_FOUND, message: `user not found`, data: null}
+    if (!user) return {success: false, status: httpStatus.NOT_FOUND, message: `user not found`};
 
     let shouldError = false;
-    let errors: any = [];
-
-    //get product names only based if quantity in >= 1 //to be fetched from newStock
+    let errors: string[] = [];
+
+    //get product ID/names only  if quantity in >= 1 //to be fetched from newStock
+    await Promise.all(
+        inStock.map( async (item) => {
+            const product = await ProductModel.findById(item.productID)
+            const newStocks = await NewStockModel.find()
```

Comment:
isn't this costly? if you do it every time

Reply:
I will have a second look into this